### ISSUE TITLE
Incorrect escaping in C++ generated code

### ISSUE THREAD

**Issue by bernardnormier** [zeroc-ice/ice#1617]
The generated code for test Slice/escape/Key.ice does not compile if you make the local interface in that file non-local.

Bogus code:
```cpp
/// \cond INTERNAL
bool
_cpp_and::_cpp_friend::_iceD_goto(::IceInternal::Incoming& inS, const ::Ice::Current& current)
{
    _iceCheckMode(::Ice::Normal, current.mode);
    ::Ice::InputStream* istr = inS.startReadParams();
    _cpp_continue iceP_if;
    _cpp_auto iceP_d;
    deletePtr iceP_inline;
    ::and::switchPtr iceP_private;    <-- should be _cpp_and
    ::and::doPtr iceP_mutable;


```

**Comment by bernardnormier** [comment]
There is a similar issue with the code generated by slice2swift.

**Comment by bernardnormier** [comment]
I verified this bug is still present.

**Comment by InsertCreativityHere** [comment]
The C++ bugs were fixed in #2786 and the Swift bugs were fixed in #2820.
